control-plane-operator/controllers/hostedcontrolplane/oauth/params.go

```diff
@@ -68,6 +88,16 @@ func NewOAuthServerParams(hcp *hyperv1.HostedControlPlane, images map[string]str
 	default:
 		p.Replicas = 1
 	}
+	if hcp.Annotations != nil {
```

Comment:
A nil map behaves like an empty map when reading so wouldn't `hcp.Annotations[hyperv1.IdentityProviderOverridesAnnotation];` be !ok if `hcp.Annotations` is nil? so `if hcp.Annotations != nil` check is redundant.

Reply:
I am fairly confident go panics if you try and index into a nil map hence that initial check but that's an easy thing to test! Will test

Reply:
fixed